security/pkg/server/ca/server.go

```diff
@@ -52,10 +51,48 @@ type Server struct {
 	monitoring     monitoringMetrics
 }
 
+// CreateCertificate handles an incoming certificate signing request (CSR). It does
+// proper validation (e.g. authentication) and upon validated, signs a certificate that:
+// the SAN is the identity of the caller in authentication result.
+// the subject public key is the public key in the CSR.
+// it is signed by the CA signing key.
+func (s *Server) CreateCertificate(ctx context.Context, request *pb.IstioCertificateRequest) (
+	*pb.IstioCertificateResponse, error) {
+	caller := s.authenticate(ctx)
+	if caller == nil {
+		log.Warn("request authentication failure")
+		s.monitoring.AuthnError.Inc()
+		return nil, status.Error(codes.Unauthenticated, "request authenticate failure")
+	}
+
+	// TODO: Call authorizer.
+
+	_, _, certChainBytes, rootCertBytes := s.ca.GetCAKeyCertBundle().GetAll()
+	cert, signErr := s.ca.Sign(
+		[]byte(request.Csr), caller.identities, time.Duration(request.ValidityDuration)*time.Second, false)
+	if signErr != nil {
```

Comment:
put a default value if ```request.ValidityDuration``` is optional ?

Reply:
Good catch. This is to be catched inside ca.Sign(). I have added the logic to check the cert lifetime.